### ISSUE TITLE
Remove DynamicChangeDetector from pre-compiled apps

### ISSUE THREAD

**Issue by yjbanov** [angular/angular#3459]
`DynamicChangeDetector` is not used in pre-compiled apps and only adds to code size.

**Comment by vsavkin** [comment]
The dynamic change detection is used when creating host views. Since the way we create host views will change by @tbosch's refactoring, I am putting this issue on hold.

**Comment by yjbanov** [comment]
Marking as blocked by https://github.com/angular/angular/issues/3605

**Comment by tbosch** [comment]
Blocked via #6301

**Comment by tbosch** [comment]
Or rather this will be solved via #6301
